### PR TITLE
Add dialog to confirm manually starting a cron job

### PR DESCRIPTION
**Pull request by fichtnerma**
---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-1007
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>
Dialog: 

image

</details>

### CODE REVIEW

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 1]
```diff
@@ -66,11 +66,17 @@ export function CronJobsGrid() {
     const intl = useIntl();
     const client = useApolloClient();
     const stackSwitchApi = useStackSwitchApi();
+    const [isOpen, setIsOpen] = useState(false);
+    const [cronJob, setCronJob] = useState("");
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:70]: You could remove the isOpen state by using the cron job to start as toggle for the dialog:

```ts
const [cronJobToStart, setCronJobToStart] = useState<string>();
const dialogOpen = Boolean(cronJobToStart);
```

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 2]
```diff
@@ -148,6 +151,31 @@ export function CronJobsGrid() {
                 disableColumnSelector
                 components={{ Toolbar: CronJobsToolbar }}
             />
+            <Dialog open={isOpen} onClose={toggleDialog}>
+                <div>
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:155]: This div shouldn't be necessary?

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 3]
```diff
@@ -148,6 +151,31 @@ export function CronJobsGrid() {
...
+                    <DialogTitle>
+                        <FormattedMessage id="comet.pages.cronjob.dialog.title" defaultMessage="Start CronJob now?" />
+                    </DialogTitle>
+                    <DialogContent>
+                        <FormattedMessage id="comet.pages.cronjob.dialog.content" defaultMessage="Are you sure you want to start the CronJob now?" />
+                    </DialogContent>
+                    <DialogActions>
+                        <CancelButton onClick={toggleDialog} />
+
+                        <SaveButton
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:165]: I don't think SaveButton is correct here: Shouldn't this be a normal Button"?

File: .changeset/friendly-parents-divide.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-admin": minor
+---
+
+Add Dialog to confirm running a CronJob
```
Comment (johnnyomair) [inline .changeset/friendly-parents-divide.md:5]:
```suggestion
Add dialog to confirm manually starting a cron job
```

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 5]
```diff
@@ -148,6 +151,31 @@ export function CronJobsGrid() {
                 disableColumnSelector
                 components={{ Toolbar: CronJobsToolbar }}
             />
+            <Dialog open={isOpen} onClose={toggleDialog}>
+                <div>
+                    <DialogTitle>
+                        <FormattedMessage id="comet.pages.cronjob.dialog.title" defaultMessage="Start CronJob now?" />
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:157]:
```suggestion
                        <FormattedMessage id="comet.pages.cronjob.dialog.title" defaultMessage="Start cron job now?" />
```

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 6]
```diff
@@ -148,6 +151,31 @@ export function CronJobsGrid() {
...
+                    </DialogTitle>
+                    <DialogContent>
+                        <FormattedMessage id="comet.pages.cronjob.dialog.content" defaultMessage="Are you sure you want to start the CronJob now?" />
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:160]:
```suggestion
                        <FormattedMessage id="comet.pages.cronjob.dialog.content" defaultMessage="Are you sure you want to start the cron job now?" />
```

File: packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx [hunk 7]
```diff
@@ -148,6 +149,31 @@ export function CronJobsGrid() {
                 disableColumnSelector
                 components={{ Toolbar: CronJobsToolbar }}
             />
+            <Dialog open={dialogOpen} onClose={closeDialog}>
+                <DialogTitle>
+                    <FormattedMessage id="comet.pages.cronjob.dialog.title" defaultMessage="Start cron job now?" />
+                </DialogTitle>
+                <DialogContent>
+                    <FormattedMessage id="comet.pages.cronjob.dialog.content" defaultMessage="Are you sure you want to start the cron job now?" />
```
Comment (dkarnutsch) [inline packages/admin/cms-admin/src/cronJobs/CronJobsGrid.tsx:157]: I would add the name of the cron job to start here

**Review by dkarnutsch — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**